src/Language/PureScript/Sugar.hs

```diff
@@ -68,4 +68,5 @@ desugar = map removeSignedLiterals
           >=> desugarImports
           >=> rebracket
           >=> desugarTypeClasses
+          >=> desugarCasesModule -- TODO: avoid duplicating this
```

Comment:
Yeah, perhaps - I barely remember what I was doing now. It's not too complicated though, I'm just not sure where the additional cases came from off the top of my head.